// gatsby-remark-instagram-embed/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var visit = require('unist-util-visit');

var embedOptions = {
  width: 320,
  height: 320
};

function instagramEmbed() {
  return transformer;
}

function embedCode(code) {
  return "<div><iframe src=\"https://instagram.com/p/" + code + "/embed\" frameborder=\"0\" allowfullscreen scrolling=\"no\" allowtransparency width=\"" + embedOptions.width + "\" height=\"" + embedOptions.width + "\"></iframe></div>";
}

function transformer(tree) {
  visit(tree, 'image', visitor);
}

function visitor(node, options) {
  var _node$alt = node.alt,
      alt = _node$alt === void 0 ? '' : _node$alt,
      _node$url = node.url,
      url = _node$url === void 0 ? '' : _node$url;

  if (alt === 'instagram') {
    node.type = 'html';
    node.value = embedCode(url);
  }
}

module.exports = function (_ref, options) {
  var markdownAST = _ref.markdownAST;

  if (options === void 0) {
    options = embedOptions;
  }

  embedOptions = (0, _extends2.default)({}, embedOptions, options);
  instagramEmbed()(markdownAST);
  return markdownAST;
};
